apps/drec-api/src/pods/file/file.service.ts

```diff
@@ -41,33 +45,44 @@ export class FileService {
           filename: this.generateUniqueFilename(file.originalname),
           data: file.buffer,
           contentType: file.mimetype,
-          userId: user.id.toString(),
-          organizationId: user.organizationId?.toString(),
+          userId: user?.id?.toString(),
+          organizationId: user?.organizationId?.toString(),
           isPublic,
```

Comment:
Basically, here you have contrary data.

so isPublic stands in opposite to `userId` (when userId is set, then isPublic should be false, and vice versa). If my understanding is good, I would remove isPublic flag, and use `userId: null` to differentiate public file from not public one

Reply:
Actually the user should not be optional. Because the method is restricted to authenticated users. So The problem is making the user optional. I will change this.